code_challenges.rb

```diff
@@ -1,48 +1,130 @@
 # ASSESSMENT 4: Ruby Coding Practical Questions
 
+puts "
+Week 4 Assessment - Ruby Code Challenges - Brian Thomas Hammond
+"
+
 # --------------------1) Create a method that takes in an array and returns an array with only odd numbers sorted from least to greatest.
 
 fullArr1 = [4, 9, 0, '7', 8, true, 'hey', 7, 199, -9, false, 'hola']
 # Expected output: [-9, 7, 9, 199]
 fullArr2 = ['hello', 7, 23, -823, false, 78, nil, '67', 6, 'Number']
 # Expected output: [-823, 7, 23]
 
+#Write a method that takes an array
+def odds_small_to_big (array) 
+    #Use .select, .is_a? Numeric, and .is_odd? to pull odd numbers
+    #Use .sort to sort the numbers, return implicitly
+    p array.select {|item| item.is_a? Numeric and item.odd?}.sort
+end
 
 
-
+puts "
+#1:
+"
+odds_small_to_big(fullArr1)# Expected output: [-9, 7, 9, 199]
+odds_small_to_big(fullArr2)# Expected output: [-823, 7, 23]
 
 # --------------------2) Create a class called Bike that is initialized with a model, wheels, and a frame size. The default number of wheels is 2. Create a get_info method that returns a sentence with all the data from the bike object.
 
-
-
-
+#Write a class named Bike
+class Bike
+    #Use attr_accessor to provide getter methods 
+    attr_accessor :model, :frame_size, :wheels, :speedometer
+    #Initialize to define model, frame size and wheels (default 2)
+    def initialize (model, frame_size, wheels = 2)
+        @model = model
+        @frame_size = frame_size
+        @wheels = wheels
+        @speedometer = 0
+    end
```

Comment:
If you are initializing wheels at 2, you might as well just set it as 2 on line 38 rather than passing it as an argument to initialize.